sky/packages/sky/lib/src/rendering/debug.dart

```diff
@@ -30,3 +32,34 @@ bool debugPaintBoundsEnabled = false;
 
 /// The color to use when painting RenderError boxes in checked mode.
 ui.Color debugErrorBoxColor = const ui.Color(0xFFFF0000);
+
+/// Prints a message to the console, which you can access using the "flutter"
+/// tool's "logs" command ("flutter logs").
+///
+/// This function very crudely attempts to throttle the rate at which messages
+/// are sent to avoid data loss on Android. This means that interleaving calls
+/// to this function (directly or indirectly via [debugDumpRenderTree] or
+/// [debugDumpApp]) and to the Dart [print] method can result in out-of-order
+/// messages in the logs.
+void debugPrint(String message) {
+  _debugPrintBuffer.addAll(message.split('\n'));
+  if (!_debugPrintScheduled)
+    _debugPrintTask();
+}
+int _debugPrintedCharacters = 0;
+int _kDebugPrintCapacity = 32 * 1024;
+Queue<String> _debugPrintBuffer = new Queue<String>();
+bool _debugPrintScheduled = false;
+void _debugPrintTask() {
+  _debugPrintScheduled = false;
+  while (_debugPrintedCharacters < _kDebugPrintCapacity && _debugPrintBuffer.length > 0) {
+    String line = _debugPrintBuffer.removeFirst();
+    _debugPrintedCharacters += line.length; // TODO(ianh): Use the UTF-8 byte length instead
+    print(line);
+  }
+  if (_debugPrintBuffer.length > 0) {
+    _debugPrintScheduled = true;
+    _debugPrintedCharacters = 0;
```

Comment:
Maybe keep a stopwatch for the last actual print and reset _debugPrintedCharacters if it's been long enough?  The way you've written this code, you'll eventually have to pay a one second pause even if you print at a slow rate.